sylvia-derive/src/parser.rs

```diff
@@ -82,7 +82,7 @@ impl MsgType {
     }
 
     /// Emits type which should be returned by dispatch function for this kind of message
-    pub fn emit_result_type(self, msg_type: &Path, err_type: &Type) -> TokenStream {
+    pub fn emit_result_type(self, msg_type: impl ToTokens, err_type: &Type) -> TokenStream {
```

Comment:
Not sure about that - now you loos your type check that the entity passed here is something reasonable. I can pass the expression here. Logically - only `Path` and maybe a `Type` or an `Ident` can go here, but you can always make both `Ident` and `Type` into `Path` with `Into` traits.